openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java

```diff
@@ -100,30 +104,95 @@ public OpenAPIConverterService() {
     }
 
     /**
-     * This API is used to return the List of {@code OASResult} that containing all the service node generated yaml
-     * content, json content, openapi content and all diagnostics and error message if generation failed.
+     * This API is used to return the Json Array of {@code OASResult} that containing all the service node generated
+     * yaml content, json content, openapi content and all diagnostics and error message if generation failed.
      */
     @JsonRequest
     public CompletableFuture<OpenAPIConverterResponse> generateOpenAPI(OpenAPIConverterRequest request) {
         return CompletableFuture.supplyAsync(() -> {
             OpenAPIConverterResponse response = new OpenAPIConverterResponse();
-            String fileUri = request.getDocumentFilePath();
-            Optional<SyntaxTree> syntaxTree = getPathFromURI(fileUri).flatMap(workspaceManager::syntaxTree);
-            Optional<SemanticModel> semanticModel = getPathFromURI(fileUri).flatMap(workspaceManager::semanticModel);
-            if (semanticModel.isEmpty() || syntaxTree.isEmpty()) {
-                StringBuilder errorString = getErrorMessage(syntaxTree, semanticModel);
-                response.setError(errorString.toString());
-            } else {
-                response.setError(null);
-                List<OASResult> oasResult = ServiceToOpenAPIConverterUtils.generateOAS3Definition(syntaxTree.get(),
-                        semanticModel.get(), null, false, null);
-                //Response handle with returning list of {@code OASResult} model.
-                response.setContent(oasResult);
+            Path filePath = Path.of(request.getDocumentFilePath());
+            Optional<SemanticModel> semanticModel = workspaceManager.semanticModel(filePath);
+            Optional<Module> module = workspaceManager.module(filePath);
```

Comment:
Does this always return the default module? According to the API description, it returns the module in the given path. If so, when the path is given for a ballerina file in another module (other than default), it'll return that specific module. Can you check on this?

Reply:
Fixed